### PR TITLE
Added a data/migration validation utility

### PR DESCRIPTION
**Pull request by Ceiu**
- Added the "cmv" tool to server/bin. This tool can be used to create
  snapshots for one or more orgs and then validate a deployment against
  a previous snapshot at a later date
- Fixed a few typos and cryptic bits of code in cpc

### CODE REVIEW

**Comment by komidore64** [comment]
protip: don't use `and` and `or` in conditional statements or boolean logic. `and` and `or` operate differently from `&&` and `||`. Always stick with `&&` and `||`.

**Comment by komidore64** [comment]
you're doing some unneeded work in all the places where you're checking for `if !some_object.nil?`. Anything other than `nil` or `false` evaluates to `true` in a `if some_object` conditional, so you can just say `if some_object`.

for instance:
```ruby
def check(obj)
  if obj
    true
  else
    false
  end
end

check(42) # => true
check("42") # => true
check(0) # even zero! => true
check('') # => true
check(nil) # => false
check(false) # => false
```

File: server/bin/cmv [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+            'orgs.pools.derivedProductAttributes' => 'name',
+            'orgs.subscriptions' => 'id',
+        }
+
+        exclude = [
+            'orgs.pools.productAttributes.updated',
+            'orgs.products.attributes.updated',
+            'orgs.products.updated',
+            'orgs.products.productContent.content.updated'
+        ]
```
Comment (dgoodwin) [inline server/bin/cmv]: Are we going to exclude the things we know will appear different? (href, owner links on products, etc) Ideally I'd like to see it run without args and return that everything is fine. Although I guess that does influence the generic nature of this a bit...

File: server/bin/cmv [hunk 2]
```diff
@@ -0,0 +1,446 @@
+#!/usr/bin/env ruby
```
Comment (dgoodwin) [inline server/bin/cmv]: Could you add a little documentation here at the top just to clarify what this is for anyone who looks in future, cmv could be a bit ambiguous.

**Pull request merged**